shell_script.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
```

Comment:
@Bentleysb mentioned this a little farther up, but I'd like to echo that it's a little out of the norm to be writing a shell script that calls one python script which writes to std out so another python script can read std in. Could we instead do this all in python, totally eliminating the need for a shell script?

Reply:
Ditto.

Reply:
Nachtm and Savannah,

I appreciate you guys suggest. The way I used the shell script has the following considerations. One is memory usage. The other is integrating other command-line tools into the system. For memory usage, my understanding is that a pipeline implementation enables the program process large amounts of data (e.g., millions or hundred millions of data). By contrast, encapsulating everything in a program might cause memory exceptions. That is the reason why I choose the shell implementation.

My understanding might be incorrect. Do you have other thoughts

Reply:
@Li-Yun Essentially, the geoProcess.py and gridGeneration.py are two standalone python programs which are currently connected via stdout and stdin. By setting it up this way, it means that each program has to serialize the output to a string to communicate. This is likely pretty inefficient, and limits the interaction possible between the two. 

By refactoring to use classes and data structures, we could leverage shared functions instead.

Reply:
remove the shell script and treat the geoProcess script as a module. The main script is able to call a parsing function in the module and return a result.